crates/consensus/beacon/src/engine/event.rs

```diff
@@ -15,6 +15,8 @@ pub enum BeaconConsensusEngineEvent {
     ForkchoiceUpdated(ForkchoiceState, ForkchoiceStatus),
     /// A block was added to the fork chain.
     ForkBlockAdded(Arc<SealedBlock>, Duration),
+    /// A parent block that extends the canonical chain
+    CanonicalParentBlockAdded(Arc<SealedBlock>, Duration),
     /// A block was added to the canonical chain, and the elapsed time validating the block
     CanonicalBlockAdded(Arc<SealedBlock>, Duration),
```

Comment:
on a second thought, we can just change this to an Option, then we don't need the additional variant entirely

Reply:
so the variant should be `CanonicalBlockAdded(Arc<SealedBlock>, Duration, Option<Arc<SealedBlock>>),` right?

Reply:
like CanonicalBlockAdded(Arc<SealedBlock>, Option<Duration>)